test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -44,32 +42,215 @@ public void ToOtlpLogRecordTest()
+
+            otlpLogRecordAttributes = otlpLogRecord.Attributes.ToString();
+
+            // Event
+            Assert.Contains("Id", otlpLogRecordAttributes);
+            Assert.Contains("10", otlpLogRecordAttributes);
+            Assert.Contains("Name", otlpLogRecordAttributes);
+            Assert.Contains("MyEvent10", otlpLogRecordAttributes);
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordTraceIdSpanIdFlagWithDroppedActivity()
+        {
+            List<LogRecord> logRecords = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.AddInMemoryExporter(logRecords);
+                });
+            });
+
+            var logger = loggerFactory.CreateLogger("OtlpLogExporterTests");
+            logger.LogInformation("Log within a dropped activity");
+            var logRecord = logRecords[0];
+            var otlpLogRecord = logRecord.ToOtlpLog();
+
+            Assert.Null(Activity.Current);
+            Assert.True(otlpLogRecord.TraceId.IsEmpty);
+            Assert.True(otlpLogRecord.SpanId.IsEmpty);
+            Assert.True(otlpLogRecord.Flags == 0);
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordSpanIdTraceIdAndFlag()
+        {
+            // preparation
+            var exportedActivityList = new List<Activity>();
+            var activitySourceName = "toOtlpLogRecordTest";
+            var activitySource = new ActivitySource(activitySourceName);
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource(activitySourceName)
+                .SetSampler(new AlwaysOnSampler())
+                .AddInMemoryExporter(exportedActivityList)
+                .Build();
+
+            using var activity = activitySource.StartActivity("Activity");
+            List<LogRecord> logRecords = new List<LogRecord>();
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.AddInMemoryExporter(logRecords);
+                });
+            });
+
+            var logger = loggerFactory.CreateLogger("OtlpLogExporterTests");
+
+            // execution
+            logger.LogInformation("Log within activity marked as RecordOnly");
+
+            // assertion
+            var logRecord = logRecords[0];
+            var otlpLogRecord = logRecord.ToOtlpLog();
+
+            var currentActivity = Activity.Current;
+            Assert.NotNull(Activity.Current);
+
+            var expectedTraceId = currentActivity.TraceId;
+            var expectedSpanId = currentActivity.SpanId;
+            Assert.Equal(expectedTraceId.ToString(), ActivityTraceId.CreateFromBytes(otlpLogRecord.TraceId.ToByteArray()).ToString());
```

Comment:
Curious - would `Assert.Equal(expectedTraceId.ToString(), otlpLogRecord.TraceId.ToString())` work? (or general question - is it possible to simplify this).

Reply:
Maybe @alanwest could help to shed some light on this too :grinning: ?

Reply:
It appears there's a `ToStringUtf8` method on `ByteString` that might get you the representation you want https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/byte-string#class_google_1_1_protobuf_1_1_byte_string_1a34312693c1a26482efd2aea46d8d4902.

Reply:
Yes, I've tried that. But ByteString method .ToStringUtf8() does not seem to perform as expected.
![image](https://user-images.githubusercontent.com/16979322/154542190-20550edf-c049-4875-bf22-4dc5a5039bb2.png)

Reply:
Maybe it's ToString(Encoding.UTF16) - just need to find out the actual encoding I guess?

Reply:
![image](https://user-images.githubusercontent.com/16979322/154552162-a901f6d7-3da1-4bd1-a0ed-fb2bb914200f.png)
Hmm... it's not Unicode either.